multifidelity_end2end.py

```diff
@@ -117,25 +141,61 @@ def main():
+        hf_test_data = [data.MoleculeDatapoint(smi, t) for smi, t in zip(hf_test_smiles, hf_test_targets)]
+        lf_test_data = [data.MoleculeDatapoint(smi, t) for smi, t in zip(lf_test_smiles, lf_test_targets)]
+
     else:
         # Initializing the data
         train_data = [data.MoleculeDatapoint(smi, t) for smi, t in zip(train_smiles, train_targets)]
         val_data = [data.MoleculeDatapoint(smi, t) for smi, t in zip(val_smiles, val_targets)]
         test_data = [data.MoleculeDatapoint(smi, t) for smi, t in zip(test_smiles, test_targets)]
 
     mgf = featurizers.MoleculeFeaturizer()
-    train_dset = data.MoleculeDataset(train_data, mgf)
-    val_dset = data.MoleculeDataset(val_data, mgf)
-    test_dset = data.MoleculeDataset(test_data, mgf)
 
-    if args.scale_data:
-        train_scaler = train_dset.normalize_targets()
-        _ = val_dset.normalize_targets(train_scaler)
-        test_scaler = test_dset.normalize_targets()
+    if args.model_type == "transfer":
+        hf_train_dset = data.MoleculeDataset(hf_train_data, mgf)
+        lf_train_dset = data.MoleculeDataset(lf_train_data, mgf)
+        hf_val_dset = data.MoleculeDataset(hf_val_data, mgf)
+        lf_val_dset = data.MoleculeDataset(lf_val_data, mgf)
+        hf_test_dset = data.MoleculeDataset(hf_test_data, mgf)
+        lf_test_dset = data.MoleculeDataset(lf_test_data, mgf)
 
-    train_loader = data.MolGraphDataLoader(train_dset, batch_size=50, num_workers=12)
-    val_loader = data.MolGraphDataLoader(val_dset, batch_size=50, num_workers=12, shuffle=False)
-    test_loader = data.MolGraphDataLoader(test_dset, batch_size=50, num_workers=12, shuffle=False)
+        if args.scale_data:
+            # Is this correct?
+            train_scaler = hf_train_dset.normalize_targets()
+            _ = hf_val_dset.normalize_targets(train_scaler)
+            test_scaler = hf_test_dset.normalize_targets()
+
+
+        hf_train_loader = data.MolGraphDataLoader(hf_train_dset, batch_size=50, num_workers=12)
+        lf_train_loader = data.MolGraphDataLoader(lf_train_dset, batch_size=50, num_workers=12)
+        hf_val_loader = data.MolGraphDataLoader(hf_val_dset, batch_size=50, num_workers=12, shuffle=False)
+        lf_val_loader = data.MolGraphDataLoader(lf_val_dset, batch_size=50, num_workers=12, shuffle=False)
+        hf_test_loader = data.MolGraphDataLoader(hf_test_dset, batch_size=50, num_workers=12, shuffle=False)
+        # Not needed?
+        lf_test_loader = data.MolGraphDataLoader(lf_test_dset, batch_size=50, num_workers=12, shuffle=False)
+
+
+    else:
+        train_dset = data.MoleculeDataset(train_data, mgf)
+        val_dset = data.MoleculeDataset(val_data, mgf)
+        test_dset = data.MoleculeDataset(test_data, mgf)
+
+        if args.scale_data:
+            train_scaler = train_dset.normalize_targets()
+            _ = val_dset.normalize_targets(train_scaler)
+            test_scaler = test_dset.normalize_targets()
```

Comment:
I'm not sure why I wrote it this way originally - the test set should also be scaled using the train scaler. Can you help me update the later parts of the script that use the `test_scaler` to use `train_scaler` instead?
```suggestion
            _ = test_dset.normalize_targets(train_scaler)
```